src/app/components/navigation/ErrorClear.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import { clearError } from "../../state";
+
+// This component clears the error state when the location changes (i.e. when the user navigates to a new page)
+
+const ErrorClear = () => {
```

Comment:
It may be worth adding a couple of tests for this (one to verify that the error state is cleared when the location changes, and another one to ensure that the error state is not cleared if the location is the same). I'm not sure how easy it is to test components that use React Router, but since it's a popular library hopefully there should be some examples online.

Reply:
I've added an integration test that checks for error clearing between page navigations, and makes sure that the function is called the expected number of times (3 route changes = 3 x clearError()).  

Not really sure on testing that it doesn't clear when location is the same? If the page was refreshed that would intentionally clear the error from state. And if it was running at other times, the number of function calls wouldn't add up to the number of route changes.

Reply:
The test for the same location could be done with a unit test where you mock `clearError` and `useLocation`. But to be honest, don't worry too much, the test you've done is enough, it covers the main use case.